File: doc-examples/azure-cosmos-example-kotlin/src/main/kotlin/example/Book.kt

package example

import io.micronaut.data.annotation.*
import java.util.*

// tag::book[]
@MappedEntity
data class Book(@field:Id
                @GeneratedValue
                var id: String?,
                var title: String,
                var pages: Int = 0,
                @MappedProperty(converter = ItemPriceAttributeConverter::class)
                var itemPrice: ItemPrice? = null,
                @DateCreated
                var createdDate: Date? = null,
                @DateUpdated
                var updatedDate: Date? = null)
// end::book[]

File: doc-examples/azure-cosmos-example-python/src/main/python/example/Book.py
# tag::book[]
from dataclasses import dataclass
from typing import Annotated

from java.util import Date
from micronaut.data.annotation import DateCreated, DateUpdated, GeneratedValue, Id, MappedEntity, MappedProperty
from micronaut.data.cosmos.annotation import PartitionKey

from example.ItemPrice import ItemPrice
from example.ItemPriceAttributeConverter import ItemPriceAttributeConverter


@MappedEntity
@dataclass
class Book:
    title: str
    pages: int
    itemPrice: Annotated[ItemPrice | None, MappedProperty(converter=ItemPriceAttributeConverter)] = None
    createdDate: Annotated[Date | None, DateCreated] = None
    updatedDate: Annotated[Date | None, DateUpdated] = None
    id: Annotated[str | None, Id, GeneratedValue, PartitionKey] = None
    # end::book[]
    # tag::book[]
    # ...
# end::book[]
